root/static/scripts/edit/URLCleanup.js

```diff
@@ -1990,6 +2072,45 @@ const CLEANUPS: CleanupEntries = {
       return {result: false, target: ERROR_TARGETS.URL};
     },
   },
+  'deviantart': {
+    match: [new RegExp('^(https?://)?(www\\.)?deviantart\\.com', 'i')],
+    restrict: [LINK_TYPES.artgallery],
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?deviantart\.com\/([^\/]+).*$/, 'https://www.deviantart.com/$1');
+      return url;
+    },
+    validate: function (url) {
+      const hardcodedPaths = [
+        'about',
+        'core-membership',
+        'forum',
+        'groups',
+        'join',
+        'shop',
+        'team',
+        'users',
+      ];
+      const m = /^https:\/\/www\.deviantart\.com\/([^\/]+)$/.exec(url);
```

Comment:
To match the suppresion of query and fragment as per previous comment:
```suggestion
      const m = /^https:\/\/www\.deviantart\.com\/([^\/?#]+)$/.exec(url);
```